### PR TITLE
Issue/11167 stats custom range hide visitor stats

### LINKED ISSUE #11167: [Stats Custom Range] Hide visitor stats when using a custom range

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #11167 

### Description
This PR adds logic for hiding the visitor stats and conversion rate values when using custom ranges in some specific cases:
1. Hides the hourly visitor count as the API doesn't support the hourly granularity.
2. Hides the total visitor count when the custom range spans over multiple days, as the manual calculation won't be accurate.
3. Shows a dialog that explain why the total visitor count can't be shown for the case 2.

The PR also fixes a bug, see https://github.com/woocommerce/woocommerce-android/pull/11169#discussion_r1539409080

##### Jetpack Site
1. Open the app.
2. Create a custom range with a single or two days.
3. Confirm the visitor count and conversion rate are shown.
4. Tap on an hour in the chart.
5. Confirm the visitor stats are hidden.
6. Modify the custom range to include more than 2 days.
7. Confirm the visitor stats are hidden, and an `info` icon is shown.
8. Tap on the visitor stats empty view or the icon.
9. Confirm this shows a dialog that explains why the stats are hidden.
10. Tap on a point in the graph.
11. Confirm the visitor stats are shown.

##### Non-jetpack site
1. Sign in using a Jetpack CP site (or using site credentials).
2. Confirm there is no regression.
3. Confirm the https://github.com/woocommerce/woocommerce-android/pull/11169#discussion_r1539409080 below is fixed.

### Images/gif
https://github.com/woocommerce/woocommerce-android/assets/1657201/0266b71f-2ad4-4ea2-8a24-48e77ff979c1

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/203">18.0</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreStatsView.kt [hunk 1]
```diff
@@ -141,6 +144,10 @@ class MyStoreStatsView @JvmOverloads constructor(
     private val chartUserInteractions = MutableSharedFlow<Unit>()
     private lateinit var chartUserInteractionsJob: Job
 
+    private var isChartValueSelected = false
+
+    private var isJetpackVisitorStatsUnavailable = false
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreStatsView.kt:149]: This property is not related to the main task of the PR, it's a fix to a bug that I identified while working on this:

https://github.com/woocommerce/woocommerce-android/assets/1657201/2e4c977a-9b7e-4f3f-8906-03b65eb16e02

**Review by 0nko — APPROVED** [review]
Everything looking good 👍. Although unrelated, I noticed that the Analytics hub still displays the visitor stats even for custom range. But that can be taken care of in a separate PR.

**Comment by hichamboushaba** [comment]
> Although unrelated, I noticed that the Analytics hub still displays the visitor stats even for custom range. But that can be taken care of in a separate PR.

Yes, this is not related to our changes, I think the team is planning to work on it in https://github.com/woocommerce/woocommerce-android/issues/10946
